### PR TITLE
Objectify 6.0

### LINKED ISSUE #3073: Objectify 6.0
Objectify 6.0 is out. Should we recommend this version to users in our wizards? Does it work on App Engine?

https://github.com/objectify/objectify/wiki/UpgradeVersion5ToVersion6

The release notes seem to reference Cloud SDK. I'm unclear why this matters to Objectify.

Objectify v6 does work with Flexible, and we seem to be picking up 6.0 automatically anyway. If we do want version 5, we have to pin the version.

### PR DESCRIPTION
**Pull request by elharo**
fix #3073 @briandealwis @chanseokoh

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/AppEngineWizardPage.java [hunk 1]
```diff
@@ -96,6 +88,13 @@ public void createControl(Composite parent) {
     Dialog.applyDialogFont(container);
   }
 
+  private void configureLibrariesSelectorGroup(Composite container) {
+    appEngineLibrariesSelectorGroup = new LibrarySelectorGroup(container, getSupportedLibraries(),
+        Messages.getString("app.engine.libraries.group")); //$NON-NLS-1$
+  }
+
+  protected abstract String getSupportedLibraries();
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/AppEngineWizardPage.java]: Given this returns a `String`, I think it's good to have a Javadoc or change the name to `getSupportedLibrariesGroup`.

File: plugins/com.google.cloud.tools.eclipse.appengine.libraries/plugin.xml [hunk 2]
```diff
@@ -45,19 +45,18 @@
     
     <library
           id="objectify"
-          group="appengine"
+          group="appengine,flexible"
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.libraries/plugin.xml]: Maybe we should just promote Datastore Cloud client library on flex?

https://cloud.google.com/appengine/docs/flexible/java/migrating#datastore

The doc also says (might be out of date),

> At this time ORM libraries that available in the standard environment such as ndb and Objectify are not supported outside of the standard environment.

Comment (elharo) [reply plugins/com.google.cloud.tools.eclipse.appengine.libraries/plugin.xml]: That doc is out of date. 6.0 supports flex: "Objectify v5 and prior use the Google App Engine SDK and therefore can only be used within Google App Engine Standard. Objectify v6+ uses the new Google Cloud SDK and can be used from anywhere - GAE Standard, GAE Flex, GCE, or outside Google Cloud entirely."

File: plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/AppEngineWizardPage.java (hunk 1 above)
Comment (elharo) [reply plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/AppEngineWizardPage.java]: changed the name

**Pull request merged**